src/translate.rs

```diff
@@ -1162,38 +1180,77 @@ impl FromGlib<i32> for Option<u64> {
     }
 }
 
-/// Translate from a pointer type without taking ownership, transfer: none.
+/// Translate from a pointer type which is annotated with `transfer none`.
+/// The resulting value is referenced at least once, by the bindings.
+///
+/// The implementation of this trait should acquire a reference to the value
+/// in a way appropriate to the type, e.g. by referencing or copying.
+/// Values obtained using this trait must be properly unreferenced on `drop()`
+/// by the implementing type.
+///
+/// This is suitable for floating references, which become strong references.
+/// It is also suitable for acquiring non-gobject values, like `gchar*`.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrNone<P: Ptr>: Sized {
     unsafe fn from_glib_none(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type taking ownership, transfer: full.
+/// Translate from a pointer type which is annotated with `transfer full`.
+/// This transfers the ownership of the value to the Rust side.
+///
+/// The implementation of this trait should not acquire any references to the value.
+/// Values obtained using this trait must be properly released on `drop()`
+/// by the implementing type.
+///
+/// Because ownership can only be transferred if something is already referenced,
+/// this is unsuitable for floating references.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrFull<P: Ptr>: Sized {
     unsafe fn from_glib_full(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type by borrowing. Don't increase the refcount
+/// Translate from a pointer type by borrowing, without affecting the refcount.
+///
+/// The implementation of this trait must not acquire any references to the value.
+/// Values obtained using this trait must *not* be unreferenced on `drop()` of the returned type.
+///
+/// The purpose of this trait is to access values inside callbacks
```

Comment:
IMHO better swap this paragraph and previous here and in other traits:
firstly we talk about usage (this parts is separated now), and then talk about implementation.

Reply:
You're right, I changed the order.